### PR TITLE
[ApplicationInsights] added breaking change warnings

### PR DESCRIPTION
**Pull request by VeryEarly**
## Description

added breaking change warnings for upcoming Az.ApplicationInsights 2.0.0

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/ApplicationInsights/ApplicationInsights/ApplicationInsights/GetApplicationInsights.cs [hunk 1]
```diff
@@ -15,10 +15,12 @@
 using Microsoft.Azure.Commands.ApplicationInsights.Models;
 using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
 using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.WindowsAzure.Commands.Common.CustomAttributes;
 using System.Management.Automation;
 
 namespace Microsoft.Azure.Commands.ApplicationInsights
 {
+    [GenericBreakingChange("API version bump up to 2020-02, output type will be also updated to match the new API version", "2.0.0")]
```
Comment (BethanyZhou) [inline src/ApplicationInsights/ApplicationInsights/ApplicationInsights/GetApplicationInsights.cs:23]: The API version contains date, does it? like 2020-02?

File: src/ApplicationInsights/ApplicationInsights/ApplicationInsights/NewApplicationInsightsLinkedStorageAccount.cs [hunk 2]
```diff
@@ -52,13 +54,15 @@ public class NewApplicationInsightsLinkedStorageAccount : ApplicationInsightsBas
             Mandatory = true,
             ValueFromPipeline = true,
             HelpMessage = "PSApplicationInsightsComponent")]
+        [CmdletParameterBreakingChange("InputObject", ChangeDescription = "Parameter InputObject will be removed in upcoming Az.ApplicationInsights 2.0.0")]
```
Comment (BethanyZhou) [inline src/ApplicationInsights/ApplicationInsights/ApplicationInsights/NewApplicationInsightsLinkedStorageAccount.cs:57]: I remember InputObject is supported in generated cmdlet, could you confirm? @VeryEarly

File: src/ApplicationInsights/ApplicationInsights/ApplicationInsights/GetApplicationInsights.cs (hunk 1 above)
Comment (VeryEarly) [reply src/ApplicationInsights/ApplicationInsights/ApplicationInsights/GetApplicationInsights.cs:23]: fixed

File: src/ApplicationInsights/ApplicationInsights/ApplicationInsights/NewApplicationInsightsLinkedStorageAccount.cs (hunk 2 above)
Comment (VeryEarly) [reply src/ApplicationInsights/ApplicationInsights/ApplicationInsights/NewApplicationInsightsLinkedStorageAccount.cs:57]: this input object is the parent resource which is not supported

[the remark this answers is not in the document]
Comment (VeryEarly) [reply src/ApplicationInsights/ApplicationInsights/ApplicationInsights/NewApplicationInsightsLinkedStorageAccount.cs:57]: this input object is the parent resource which is not supported

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**